x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -598,29 +695,79 @@ export class Embeddable
     }
   }
 
+  private async updateViewUnderlyingDataArgs() {
```

Comment:
I'm not sure I quite understand. Do you mean move it to a standalone function? 

I think it initially needs to be evaluated when both `activeData` and `externalSearchContext` change because they don't necessarily resolve serially.

The function only works when both are present, but we don't know which one will resolve first, so it is called in both places and the second time it's called we know that both `activeData` and `externalSearchContext` will be available.

Reply:
IMHO it makes sense to keep it inside of the embeddable, but only call it if requested from the action (see other comment)